### ISSUE TITLE
Latest change in 2.7.4 breaks models that have an attribute `type`

### ISSUE THREAD

**Issue by YoranBrondsema** [danielspaniel/ember-data-factory-guy#233]
We have a non-polymorphic model that has an attribute `type`. This was working fine on 2.7.3 but since 2.7.4 ([this commit](https://github.com/danielspaniel/ember-data-factory-guy/commit/b015ef25dca5f03a77b147780fce1c651cd14872) in particular) the tests fail as it's trying to create a model of type `type` rather than the generic model with an attribute `type`.

More concretely, we have a model `user` that has an attribute `type`:

````
DS.Model.extend({
  type: DS.attr('string')
});
````

We then have factories for several types. So
````
FactoryGuy.define('user', {
  teacher: {
    type: 'teacher'
  },
  student: {
    type: 'student'
  }
});
````

In 2.7.3, doing `make('teacher')` resulted in creating a record of type `user` with the `type` attribute equal to `teacher`.

In 2.7.4, doing `make('teacher')` results in creating a record of type `teacher` (which fails because there's no such model).

There's probably something to be said about our choice of the attribute name `type`. But I think explicitly indicating a factory as `polymorphic` could provide a way to make a distinction between the two cases and act accordingly.

**Comment by danielspaniel** [comment]
I have been waiting for a complaint about this issue for a long time.  As you said, the latest commit forced the issue. There is a way to fix it ( I think ) .. but give me a moment to do that.

**Comment by YoranBrondsema** [comment]
Actually we do use Rails on the back-end :-). And there it's a polymorphic model. However, to keep the front-end model architecture simpler, we decided not to use the polymorphism there.

**Comment by danielspaniel** [comment]
Hmm .. wow .. that is interesting.  

Not sure actually how to fix this issue, since I can't really force people to say this factory is polymorphic ( can I ? ) .. seems pretty harsh but maybe it's not too bad?  Otherwise there is no way to tell if that type attibute is used for the polymorphic type or not .. any thoughts?

**Comment by YoranBrondsema** [comment]
That's what I was thinking about too. Polymorphism is only enabled by your latest commit right?

That would be the most "flexible" in terms of allowing everyone to configure their app to work with `ember-data-factory-guy`. I agree that our choice of naming causes confusion but we're technically not doing anything "wrong" wrt the JSON-API standard and thus Ember Data.

**Comment by danielspaniel** [comment]
Polymorphism has been enabled forever in factory guy. I just always based everything on the assumption that no one would use the type attribute. 
  So, let me ponder what to do about this. And yeah, I guess your not doing anything wrong, just bringing up a tricky issue.

**Comment by danielspaniel** [comment]
How about it I make it so you flag your factory as NOT polymorphic    
```js
polymorphic: false
```

when you are using type as attribute 

That was it is backwards compatible. And it is the exception anyway, so ?? what do you think about that.

**Comment by YoranBrondsema** [comment]
That sounds like a good solution to me!

**Comment by danielspaniel** [comment]
fixed in v2.7.5 ( but let me know if otherwise )
